### ISSUE TITLE
Modify the yt-dlp command

### ISSUE THREAD

**Issue by flossgeek** [gpodder/gpodder#1307]
I'm wondering whether it is possible to modify the yt-dlp command that is run when downloading podcasts with the Youtube-dl extension. Right now one can only modify the audio format that is being downloaded. Having the ability to add arguments to yt-dlp would be incredibly useful for applications like Sponsorblock.

**Comment by tpikonen** [comment]
gPodder uses yt-dlp as a Python module, so command line args are not possible. The [YoutubeDL class](https://github.com/yt-dlp/yt-dlp/blob/28163422a667d63cceb0e8bf27d150951e78d313/yt_dlp/YoutubeDL.py#L155) takes a decent set of parameters though, which you can set in the extension source code.

**Comment by flossgeek** [comment]
I don't see a parameter for sponsorblock in YoutubeDL class, so I assume that needs to be added by the yt-dlp guys to use it?

**Comment by auouymous** [comment]
It is technically possible but might be difficult to implement. You can add options to self._ydl_opts using `self._ydl_opts[key] = value`, the problem is splitting a config string into keys and values and then adding them to the options. Some options require a value and others don't. Maybe it only supports flag options and uses True for the value. But then you run into issues like `--no-cache-dir` on command line and `cachedir = False` in python. Some might be simple and just require removing the leading hyphens and replacing other hyphens with underscores. While others like `-F` would need to be translated to `-list-formats`.

i would be fine with an implemention to support any user-defined options. However, I would be against adding support that only removed ads from downloads. I do run UBO in the browser not to remove ads but to remove all third-party and as many first-party fetches as possible including images, CSS and javascript. Blocking third-party code and content that could be malicious is one thing, but ads embedded in media can't be anymore malicious than the media itself.

**Comment by flossgeek** [comment]
Yes, I agree that user-defined options would be a far better idea (sponsorblock was just as an example). That way yt-dlp can be used to its full potential in this implementation.

**Comment by tpikonen** [comment]
> Maybe the YoutubeDL class could load user option like the [command line](https://github.com/yt-dlp/yt-dlp/blob/4429fd0450a3fbd5e89573e06533c1a0874fae42/yt_dlp/__init__.py#L950)?

That could work. I tried to do that in #1805, please test.

**Comment by auouymous** [comment]
No, the format would be overridden after loading it from the config file. You would need to duplicate it inside gpodder.

**Comment by auouymous** [comment]
> Do I understand correctly that we have to enable the extension youtube-dl and enable both options "manage channel" and "manage downloads"?

You only need manage channel if you want more than the 15 newest episodes per channel, at the cost of very slow updates. Manage downloads is required to download from youtube, but once enabled, your audio only setting will work fine. The same settings were used for gpodder and the extension, so nothing changes when enabling manage downloads.

Try the following:

>     "preferred_fmt_id": 0,
>     "preferred_fmt_ids": ["bestaudio"],


> There are other shortcuts like adding a channel, but because I think the writing cursor doesn't really focus on the text entry box there is not really a way a blind person can enter or paste text there anymore.

I wonder if it is broken on Windows or Mac, because I can open the add channel dialog, press ctrl+v to paste and press enter to add it.


> Just a question out of curiosity: If GPodder would have to be designed new from scratch would be HTML be an option?

The 'G' in gPodder stands for gtk, no 'H' in there. ;)

But apps like you describe ship an entire Chromium browser called Electron. They are very large downloads and use a lot of memory.


> Oh, this is great, I missed that. I updated from build 1604 to 1638. But I couldn't find this update button yet. Where is it located? Would it be possible to assign a shortcut to it, maybe something like F9?

It is at the top of the youtube extension page in preferences. However, there is a report on the mailing list that the button might be broken on Windows.

And no, it isn't possible to add a shortcut key.

**Comment by computerfreak105** [comment]
Hi auouymous

 �

You wrote:

Try the following:

"preferred_fmt_id": 0,
